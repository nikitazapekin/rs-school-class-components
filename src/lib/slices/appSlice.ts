import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { AppStatee, UserDataArray, SearchTypes } from '../types';
const initialState: AppStatee = {
	isLoading: false,
	error: null,
	isLoadingUserData: false,
	users: [],
	params: {
		limit: 10,
		offset: 1,
		query: '',
		storedValue: '',
	},
};
const appSlicee = createSlice({
	name: 'app',
	initialState,
	reducers: {
		setLoading(state, action: PayloadAction<boolean>) {
			console.log('LOOO', action.payload);
			state.isLoading = action.payload;
		},
		setLoadingUserData(state, action: PayloadAction<boolean>) {
			state.isLoadingUserData = action.payload;
		},
		setError(state, action: PayloadAction<string | null>) {
			state.error = action.payload;
		},

		setSearchParamsURL(state, action: PayloadAction<SearchTypes>) {
			state.params.query = action.payload.query;
			state.params.offset = action.payload.offset;
		},
		setFirstPage(state) {
			state.params.offset = 1;
		},
		setNextPage(state) {
			state.params.offset = state.params.offset + 1;
		},
		setPrevPage(state) {
			state.params.offset = state.params.offset - 1;
		},
		setQueryPage(state, action: PayloadAction<string>) {
			state.params.query = action.payload;
		},
		setUsers(state, action: PayloadAction<UserDataArray>) {
			state.users = action.payload;
		},
		setStoredInLocalStorageQuery(state, action: PayloadAction<string>) {
			state.params.storedValue = action.payload;
			state.params.query = action.payload;
		},
	},
});
export const {
	setLoading,
	setError,
	setLoadingUserData,
	setSearchParamsURL,
	setNextPage,
	setPrevPage,
	setQueryPage,
	setUsers,
	setFirstPage,
	setStoredInLocalStorageQuery,
} = appSlicee.actions;
export default appSlicee.reducer;
