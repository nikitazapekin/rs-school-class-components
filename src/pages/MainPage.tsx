import Header from "../components/Header";
import Preview from "../components/Preview";

 
const MainPage = () => {
    return (
    <>
<Header />
 <Preview />
    </>  );
}
 
export default MainPage;