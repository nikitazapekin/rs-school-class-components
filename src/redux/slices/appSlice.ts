import { createSlice, PayloadAction, createAsyncThunk } from '@reduxjs/toolkit';
import { AppStatee, SearchTypes, User, UserDataArray } from '../types';


export interface UserData {
	total_count: number;
	incomplete_results: boolean;
	items: UserDataArray;
}


export const initialState: AppStatee = {
	isLoading: false,
	error: null,
	isLoadingUserData: false,
	users: [],
	typedValue: "",
	params: {
		limit: 10,
		offset: 1,
		query: '',
		storedValue: '',
	},
	status: 'idle',
	clickedUser: {
		login: "",
	id: 0,
	node_id:"" ,
	avatar_url: "",
	gravatar_id:"" ,
	url:"" ,
	html_url:"" ,
	followers_url: "",
	following_url:"" ,
	gists_url: "",
	starred_url: "",
	subscriptions_url:  "",
	organizations_url: "",
	repos_url:"" ,
	events_url:"" ,
	received_events_url:"" ,
	type: "",
	site_admin: false,
	score: 0
	}
   
};
const appSlicee = createSlice({
	name: 'app',
	initialState,
	reducers: {
		setLoading(state, action: PayloadAction<boolean>) {
			state.isLoading = action.payload;
		},
		setLoadingUserData(state, action: PayloadAction<boolean>) {
			state.isLoadingUserData = action.payload;
		},
		setError(state, action: PayloadAction<string | null>) {
			state.error = action.payload;
		},

		setSearchParamsURL(state, action: PayloadAction<SearchTypes>) {
			state.params.query = action.payload.query;
			state.params.offset = action.payload.offset;
		},
		setFirstPage(state) {
			state.params.offset = 1;
		},
		setNextPage(state) {
			state.params.offset = state.params.offset + 1;
		},
		setPrevPage(state) {
			state.params.offset = state.params.offset - 1;
		},
		setQueryPage(state, action: PayloadAction<string>) {
			console.log("PAYLOAD", action.payload)
			state.params.query = action.payload;
			console.log("NEW PARAMS", state.params)
		},
		setUsers(state, action: PayloadAction<UserDataArray>) {
			state.users = action.payload;
		},
		setStoredInLocalStorageQuery(state, action: PayloadAction<string>) {
			state.params.storedValue = action.payload;
			state.params.query = action.payload;
		},
		setTypedValue(state, action: PayloadAction<string>) {
		state.typedValue = action.payload
		},
		setNewSearchValue(state) {
		//	state.typedValue = action.payload
		state.params.query=state.typedValue
		state.params.storedValue=state.typedValue
		state.params.offset=1
			},
			setClickedUser(state, action: PayloadAction<User>) {
				//	state.typedValue = action.payload
			//	state.params.query=state.typedValue
			//	state.params.storedValue=state.typedValue
			state.clickedUser = action.payload
			console.log("CLICKE", JSON.stringify(state.clickedUser))
					},
	},




});
export const {
	setLoading,
	setError,
	setLoadingUserData,
	setSearchParamsURL,
	setNextPage,
	setPrevPage,
	setQueryPage,
	setUsers,
	setFirstPage,
	setStoredInLocalStorageQuery,
	setTypedValue,
	setNewSearchValue,
	setClickedUser
} = appSlicee.actions;
export default appSlicee.reducer;
