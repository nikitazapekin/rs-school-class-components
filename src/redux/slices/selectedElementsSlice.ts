import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { SelectedElementsState, UserItem } from '../types';
export const initialState: SelectedElementsState = {
	storedElements: [],

	selectedElement: null,
};
const selectedElementsSlice = createSlice({
	name: 'selectedElements',
	initialState,
	reducers: {
		setAddToStoredElement(state, action: PayloadAction<UserItem>) {
			if (!state.storedElements.some((item) => item.id == action.payload.id)) {
				state.storedElements = [...state.storedElements, action.payload];
			} else {
				state.storedElements = state.storedElements.filter((item) => item.id != action.payload.id);
			}
		},
		setRemoveStoredElementById(state, action: PayloadAction<number>) {
			state.storedElements = state.storedElements.filter((item) => item.id != action.payload);
		},
		setClearStoredElements(state) {
			state.storedElements = [];
		},
		setSelectedElement(state, action: PayloadAction<UserItem | undefined | null>) {
			if (action.payload != null && action.payload != undefined) {
				state.selectedElement = action.payload;
			}
		},
	},
});
export const { setAddToStoredElement, setClearStoredElements, setSelectedElement, setRemoveStoredElementById } =
	selectedElementsSlice.actions;
export default selectedElementsSlice.reducer;
