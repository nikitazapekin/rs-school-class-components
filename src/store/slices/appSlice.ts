import { createSlice, PayloadAction  } from '@reduxjs/toolkit';
import { AppStatee, SearchTypes,  UserDataArray } from '../types';
 import {User} from "../types"
export interface UserData {
	total_count: number;
	incomplete_results: boolean;
	items: UserDataArray;
}
export const initialState: AppStatee = {
	isLoading: false,
	error: null,
	isLoadingUserData: false,
	users: [],
	typedValue: "",
	params: {
		limit: 10,
		offset: 1,
		query: '',
		storedValue: '',
	},
	status: 'idle',
	clickedUser: {
		login: "",
	id: 0,
	node_id:"" ,
	avatar_url: "",
	gravatar_id:"" ,
	url:"" ,
	html_url:"" ,
	followers_url: "",
	following_url:"" ,
	gists_url: "",
	starred_url: "",
	subscriptions_url:  "",
	organizations_url: "",
	repos_url:"" ,
	events_url:"" ,
	received_events_url:"" ,
	type: "",
	site_admin: false,
	score: 0
	}
   
};
const appSlicee = createSlice({
	name: 'app',
	initialState,
	reducers: {
		setLoading(state, action: PayloadAction<boolean>) {
			state.isLoading = action.payload;
		},
		setLoadingUserData(state, action: PayloadAction<boolean>) {
			state.isLoadingUserData = action.payload;
		},
		setError(state, action: PayloadAction<string | null>) {
			state.error = action.payload;
		},

		setSearchParamsURL(state, action: PayloadAction<SearchTypes>) {
			state.params.query = action.payload.query;
			state.params.offset = action.payload.offset;
		},
		setFirstPage(state) {
			state.params.offset = 1;
		},
		setNextPage(state) {
			state.params.offset = state.params.offset + 1;
		},
		setPrevPage(state) {
			state.params.offset = state.params.offset - 1;
		},
		setQueryPage(state, action: PayloadAction<string>) {
			state.params.query = action.payload;
		},
		setUsers(state, action: PayloadAction<UserDataArray>) {
			state.users = action.payload;
		},
		setStoredInLocalStorageQuery(state, action: PayloadAction<string>) {
			state.params.storedValue = action.payload;
			state.params.query = action.payload;
		},
		setTypedValue(state, action: PayloadAction<string>) {
		state.typedValue = action.payload
		},
		setNewSearchValue(state) {
		state.params.query=state.typedValue
		state.params.storedValue=state.typedValue
			},
			setClickedUser(state, action: PayloadAction<User>) {
			 
			state.clickedUser = action.payload
			 
					},
	},

 

});
export const {
	setLoading,
	setError,
	setLoadingUserData,
	setSearchParamsURL,
	setNextPage,
	setPrevPage,
	setQueryPage,
	setUsers,
	setFirstPage,
	setStoredInLocalStorageQuery,
	setTypedValue,
	setClickedUser,
	setNewSearchValue,
	 
} = appSlicee.actions;
export default appSlicee.reducer;