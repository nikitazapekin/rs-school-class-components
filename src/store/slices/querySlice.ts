import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';

export const githubApi = createApi({
  reducerPath: 'githubApi',
  baseQuery: fetchBaseQuery({ baseUrl: 'https://api.github.com/' }),
  endpoints: (builder) => ({
    searchUsers: builder.query({
      query: ({ query = 'type:user', page = 1, per_page = 30 }) => ({
        url: 'search/users',
        params: {
          q: query || 'type:user',
          page,
          per_page,
        },
      }),
    }),
  }),
});

export const { useSearchUsersQuery, useLazySearchUsersQuery } = githubApi;


/*
import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';

export const githubApi = createApi({
  reducerPath: 'githubApi',  // Updated to match the correct API
  baseQuery: fetchBaseQuery({ baseUrl: 'https://api.github.com/' }),
  endpoints: (builder) => ({
    searchUsers: builder.query({
      query: ({ query, page = 1, per_page = 30 }) => ({
        url: 'search/users',
        params: {
          q: query,
          page,
          per_page
        }
      }),
    }),
  }),
});

export const { useSearchUsersQuery, useLazySearchUsersQuery } = githubApi;
*/
//`https://api.github.com/search/users?q=${typedValue}&page=${offset}&per_page=${limit}`;

/*
import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react'
 
export const githubApi = createApi({
  reducerPath: 'pokemonApi',
  // baseQuery: fetchBaseQuery({ baseUrl: 'https://pokeapi.co/api/v2/' }),
  baseQuery: fetchBaseQuery({ baseUrl: 'https://api.github.com/search/users?' }),
  endpoints: (builder) => ({
    getPokemonByName: builder.query<string, string>({
     // query: (name) => `pokemon/${name}`,
     query: (name) => ``
    }),
  }),
})
//
 
export const { useGetPokemonByNameQuery, useLazyGetPokemonByNameQuery } = githubApi

*/