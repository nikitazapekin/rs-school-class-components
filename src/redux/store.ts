import { configureStore, combineReducers } from "@reduxjs/toolkit";
 
import type { ThunkAction, Action } from '@reduxjs/toolkit';
import appSlice from "./slices/appSlice";
import selectedElementsSlice from "./slices/selectedElementsSlice";
 
import { createWrapper } from "next-redux-wrapper";
import themeSlice from "./slices/themeSlice";
export const rootReducer = combineReducers({
    appSlice: appSlice,
    selectedElementsSlice: selectedElementsSlice,
    themeSlice: themeSlice
     
});
export const store = configureStore({
 
    reducer: rootReducer,
  
   
})



export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<ReturnType, RootState, unknown, Action<string>>; 

export const wrapper = createWrapper(() => store);
