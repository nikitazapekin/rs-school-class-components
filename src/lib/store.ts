//lib/store.ts

import { combineReducers, configureStore } from '@reduxjs/toolkit';
import type { ThunkAction, Action } from '@reduxjs/toolkit';
import appSlice from './slices/appSlice';
import { setupListeners } from '@reduxjs/toolkit/query';
import { githubApi } from './slices/querySlice';
import { usersGithubApi } from './slices/userQuerySlice';
import selectedElementsSlice from './slices/selectedElementsSlice';
import testSlice from './slices/testSlice';

const rootReducer = combineReducers({
  appSlice: appSlice,
  selectedElementsSlice: selectedElementsSlice,
  testSlice: testSlice,
  [githubApi.reducerPath]: githubApi.reducer,
  [usersGithubApi.reducerPath]: usersGithubApi.reducer,
});

export const makeStore = () => {
  return configureStore({
    reducer: rootReducer,
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware().concat(githubApi.middleware).concat(usersGithubApi.middleware),
  });
};

// Infer the type of makeStore
export type AppStore = ReturnType<typeof makeStore>;
// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<AppStore['getState']>;
export type AppDispatch = AppStore['dispatch'];

// Set up listeners
const store = makeStore();
setupListeners(store.dispatch);

export default store;

/*import { configureStore } from '@reduxjs/toolkit'

export const makeStore = () => {
  return configureStore({
    reducer: {},
  })
}

// Infer the type of makeStore
export type AppStore = ReturnType<typeof makeStore>
// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<AppStore['getState']>
export type AppDispatch = AppStore['dispatch']
*/




/*
import { combineReducers, configureStore } from '@reduxjs/toolkit';
import type { ThunkAction, Action } from '@reduxjs/toolkit';
import appSlice from './slices/appSlice';
import { setupListeners } from '@reduxjs/toolkit/query';
import { githubApi } from './slices/querySlice';
import { usersGithubApi } from './slices/userQuerySlice';
import selectedElementsSlice from './slices/selectedElementsSlice';

export const rootReducer = combineReducers({
	appSlice: appSlice,
	selectedElementsSlice: selectedElementsSlice,
	[githubApi.reducerPath]: githubApi.reducer,
	[usersGithubApi.reducerPath]: usersGithubApi.reducer,
});

export const store = configureStore({
	reducer: rootReducer,
	middleware: (getDefaultMiddleware) =>
		getDefaultMiddleware().concat(githubApi.middleware).concat(usersGithubApi.middleware),
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<ReturnType, RootState, unknown, Action<string>>;

setupListeners(store.dispatch);

*/