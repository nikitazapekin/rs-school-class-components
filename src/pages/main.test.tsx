// src/pages/main.test.tsx

import { render } from '@testing-library/react';
import MainPage from './main';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import { githubApi } from '@/store/slices/querySlice';
import { usersGithubApi } from '@/store/slices/userQuerySlice';
import appSlice from '@/store/slices/appSlice';
import selectedElementsSlice from '@/store/slices/selectedElementsSlice';

const store = configureStore({
  reducer: {
    appSlice,
    selectedElementsSlice,
    [githubApi.reducerPath]: githubApi.reducer,
    [usersGithubApi.reducerPath]: usersGithubApi.reducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware()
      .concat(githubApi.middleware)
      .concat(usersGithubApi.middleware),
});

describe('MainPage', () => {
  it('should render MainPage component correctly', () => {
 
    store.dispatch = jest.fn(); // Mock dispatch

    const component = render(
      <Provider store={store}>
        <MemoryRouter>
          <MainPage />
        </MemoryRouter>
      </Provider>
    );

    expect(component).toMatchSnapshot();
  });
});


/*import { render } from '@testing-library/react';
import MainPage from './main';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { AppDispatch, RootState } from '@/store/store'; // Adjust the path as necessary

const mockStore = configureStore<RootState, AppDispatch>([]);

describe('MainPage', () => {
  it('should render App and Outlet components', () => {
    const initialState: RootState = {
      selectedElementsSlice: {
        selectedElement: null,
        storedElements: [], // Add other required properties if necessary
      },
      appSlice: {
        isLoading: false,
        error: null,
     
        users: [],
        params: {
            limit: 10,
            offset: 1,
            query: "",
            storedValue: ""
        },
      },
      [githubApi.reducerPath]: githubApi.reducer,
      [usersGithubApi.reducerPath]: usersGithubApi.reducer,  
      // Add other slices of the state if needed
    };

    const store = mockStore(initialState);

    const component = render(
      <Provider store={store}>
        <MemoryRouter>
          <MainPage />
        </MemoryRouter>
      </Provider>
    );

    expect(component).toMatchSnapshot();
  });
});

*/

/* import { render } from '@testing-library/react';
 [{
	"resource": "/c:/Users/wotbl/search-rs/src/pages/main.test.tsx",
	"owner": "typescript",
	"code": "2739",
	"severity": 8,
	"message": "Type '{ selectedElementsSlice: { selectedElement: null; storedElements: never[]; }; appSlice: { isLoading: false; error: null; users: never[]; params: { limit: number; offset: number; query: string; storedValue: string; }; }; }' is missing the following properties from type '{ appSlice: AppStatee; selectedElementsSlice: SelectedElementsState; githubApi: CombinedState<{ searchUsers: QueryDefinition<{ query: string; page: number; per_page: number; }, BaseQueryFn<...>, never, SearchUsersResponse, \"githubApi\">; }, never, \"githubApi\">; userGithubApi: CombinedState<...>; }': [githubApi.reducerPath], [usersGithubApi.reducerPath]",
	"source": "ts",
	"startLineNumber": 12,
	"startColumn": 11,
	"endLineNumber": 12,
	"endColumn": 23
}]
import MainPage from './main';
 
import { MemoryRouter } from 'react-router-dom';
// Mock the App component to ensure it's rendered correctly
//jest.mock('../App', () => () => <div data-testid="app-component">Mocked App</div>);

describe('MainPage', () => {
  it('should render App and Outlet components', () => {
 


    const component = render(
        <MemoryRouter>
           <MainPage /> 
        </MemoryRouter>
    );

    expect(component).toMatchSnapshot();

 
  });
});

*/

/*
     const component = render(
            <MemoryRouter>
                <Card
                    user={{
                        id: 1,
                        login: "Nik",
                        node_id: "1",
                        avatar_url: "dcdsc",
                        gravatar_id: "kjj",
                        url: "lmd",
                        html_url: "ldm",
                        followers_url: "ldmv",
                        following_url: "nkdv",
                        gists_url: "ndls",
                        starred_url: "cmk",
                        subscriptions_url: "kekc",
                        organizations_url: "kdc",
                        repos_url: "mld",
                        events_url: "kmd",
                        received_events_url: "dclm",
                        type: "dlcm",
                        site_admin: false,
                        score: 22
                    }}
                />
            </MemoryRouter>
        );

        expect(component).toMatchSnapshot();
        */