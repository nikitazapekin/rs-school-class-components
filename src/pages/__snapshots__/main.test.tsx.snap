// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage should render MainPage component correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <header
          class="header "
        >
          <div
            class="header__content"
          >
            <nav
              class="header__search"
            >
              <input
                class="search__bar"
                placeholder="Search..."
                type="text"
              />
              <button
                aria-label="search"
                class="search__btn"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-magnifying-glass "
                  data-icon="magnifying-glass"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
                    fill="currentColor"
                  />
                </svg>
              </button>
            </nav>
            <button
              class="redirect__btn"
            >
              Redirect to error page
            </button>
            <button
              class="theme__button"
            >
              Switch Theme 
              Light
            </button>
          </div>
        </header>
        <section
          class="list"
        >
          <div
            class="list__container"
          >
            <div
              class="list__btns"
            >
              <button
                class="list__prev list__btn"
              >
                Prev
              </button>
              <button
                class="list__next list__btn"
                data-testid="next"
              >
                Next
              </button>
            </div>
          </div>
        </section>
        <div>
          <button
            class="throw__error"
            data-testid="throw__error"
          >
            Throw Error
          </button>
        </div>
      </div>
      <div
        class="background "
        data-testid="background"
      >
           
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <header
        class="header "
      >
        <div
          class="header__content"
        >
          <nav
            class="header__search"
          >
            <input
              class="search__bar"
              placeholder="Search..."
              type="text"
            />
            <button
              aria-label="search"
              class="search__btn"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-magnifying-glass "
                data-icon="magnifying-glass"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </nav>
          <button
            class="redirect__btn"
          >
            Redirect to error page
          </button>
          <button
            class="theme__button"
          >
            Switch Theme 
            Light
          </button>
        </div>
      </header>
      <section
        class="list"
      >
        <div
          class="list__container"
        >
          <div
            class="list__btns"
          >
            <button
              class="list__prev list__btn"
            >
              Prev
            </button>
            <button
              class="list__next list__btn"
              data-testid="next"
            >
              Next
            </button>
          </div>
        </div>
      </section>
      <div>
        <button
          class="throw__error"
          data-testid="throw__error"
        >
          Throw Error
        </button>
      </div>
    </div>
    <div
      class="background "
      data-testid="background"
    >
         
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
